File: back/src/main/java/csh/back/domain/trip/group/dto/response/TripGroupResponse.java
package csh.back.domain.trip.group.dto.response;

import csh.back.domain.trip.group.entity.TripGroup;

import java.time.LocalDate;

public record TripGroupResponse(
		Long id,
		String name,
		Long ownerId,
		String region,
		String joinCode,
		int nights,
		LocalDate startDate,
		LocalDate endDate
){
	public static TripGroupResponse from(TripGroup tripGroup) {
		return new TripGroupResponse(
				tripGroup.getId(),
				tripGroup.getName(),
				tripGroup.getOwner().getId(),
				tripGroup.getRegion(),
				tripGroup.getJoinCode(),
				tripGroup.getNights(),
				tripGroup.getStartDate(),
				tripGroup.getEndDate()
		);
	}
}

File: back/src/main/kotlin/csh/back/domain/trip/group/dto/response/TripGroupResponse.kt
package csh.back.domain.trip.group.dto.response

import csh.back.domain.trip.group.entity.TripGroup
import java.time.LocalDate

data class TripGroupResponse(
    val id: Long,
    val name: String,
    val ownerId: Long,
    val region: String,
    val joinCode: String,
    val nights: Int,
    val startDate: LocalDate,
    val endDate: LocalDate,
) {
    companion object {
        @JvmStatic
        fun from(tripGroup: TripGroup): TripGroupResponse = TripGroupResponse(
            id = requireNotNull(tripGroup.id),
            name = tripGroup.name,
            ownerId = requireNotNull(tripGroup.owner.id),
            region = tripGroup.region,
            joinCode = tripGroup.joinCode,
            nights = tripGroup.nights,
            startDate = tripGroup.startDate,
            endDate = tripGroup.endDate,
        )
    }
}
